.github/workflows/documentation.yml

```diff
@@ -0,0 +1,57 @@
+# yamllint disable
+env:
+  PYTHON_PACKAGE_MANAGER: "conda" # Python package manager to create the python environments
+
+name: documentation
+
+on:
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+jobs:
+  build-documentation:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -l {0}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: conda-incubator/setup-miniconda@v3
+
+      - name: Install dependencies
+        run: |
+          $PYTHON_PACKAGE_MANAGER env create -f environment.yml
+          $PYTHON_PACKAGE_MANAGER activate queens
+          pip install -e .[develop]
```

Comment:
Since the same stuff as in the main action, I propose to split this up in a followup. What do you think?

Reply:
Do you mean moving this section to a reusable workflow file or adding the documentation to the github_ci? Because the latter would be pretty nasty with the different triggers but the first would be quite nice.

Reply:
I mean the first one, reuse existing code to create the env etc.

Reply:
Looks good to me